Orange/widgets/visualize/owscatterplot.py

```diff
@@ -380,6 +387,46 @@ def update_graph(self, attributes=None, reset_view=True, **_):
         self.graph.zoomStack = []
         if attributes and len(attributes) == 2:
             self.attr_x, self.attr_y = attributes
+        data_x = self.data[:, self.attr_x]
+        data_y = self.data[:, self.attr_y]
+        if self.attr_x not in self.attr_nans:
+            self.attr_nans[self.attr_x] = BasicStats(data_x, self.attr_x).nans
+        if self.attr_y not in self.attr_nans:
+            self.attr_nans[self.attr_y] = BasicStats(data_y, self.attr_y).nans
+        self.information()
+        self.warning()
+        flag_x = 0
+        flag_y = 0
+        if self.attr_nans[self.attr_x]:
+            flag_x = 1
+            if self.attr_nans[self.attr_x] == len(data_x):
+                flag_x = 2
+        if self.attr_nans[self.attr_y]:
+            flag_y = 1
+            if self.attr_nans[self.attr_y] == len(data_y):
+                flag_y = 2
+        if flag_x == 2:
+            if flag_y == 2:
+                self.warning("{} and {} have no defined values".format(
+                    self.attr_x, self.attr_y))
+            else:
+                self.warning("{} has no defined values".format(
+                    self.attr_x))
+        else:
+            if flag_y == 2:
+                self.warning("{} has no defined values".format(
+                    self.attr_y))
+        if flag_x == 1:
+            if flag_y == 1:
+                self.information("{} and {} have missing values".format(
+                    self.attr_x, self.attr_y))
+            else:
+                self.information("{} has missing values".format(
+                    self.attr_x))
+        else:
+            if flag_y == 1:
+                self.information("{} has missing values".format(
+                    self.attr_y))
```

Comment:
It shows the same with your code if one attribute has _all_ missing values (one warning) and some attribute has _some_ missing values (one information). If you hover over it, it shows more details in the tooltip. I don't know, I like the latter code better. @lanzagar, what do you think?